controllers/placementapi_controller.go

```diff
@@ -58,11 +62,133 @@ import (
 	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 )
 
+type conditionUpdater interface {
+	Set(c *condition.Condition)
+	MarkTrue(t condition.Type, messageFormat string, messageArgs ...interface{})
+}
+
+type GetSecret interface {
+	GetSecret() string
+	client.Object
+}
```

Comment:
If we want to move ensureSecret to lib-common then I agree to have this generalization here and the as a second step moved to lib-common. However if we won't do that move then this generalization is unnecessary in placement as in placement we have a single CR type to handle so we can inline that type directly. (In nova we use this generalization as we have multiple CR types to handle with the same code)

Reply:
Agree, I created issues for both repos https://github.com/openstack-k8s-operators/nova-operator/issues/649 and https://github.com/openstack-k8s-operators/placement-operator/issues/133 . Both I will fix in followup

Reply:
ack